dvc/stage.py

```diff
@@ -767,16 +768,22 @@ def _run(self):
         executable = os.getenv("SHELL") if os.name != "nt" else None
         self._warn_if_fish(executable)
 
-        p = subprocess.Popen(
-            self.cmd,
-            cwd=self.wdir,
-            shell=True,
-            env=fix_env(os.environ),
-            executable=executable,
-        )
-        p.communicate()
+        old_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
+        p = None
+
+        try:
+            p = subprocess.Popen(
```

Comment:
Child process is inheriting your signal mask, so you need to restore it. See my comment https://github.com/iterative/dvc/pull/2153#discussion_r295858911 where I use preexec_fn for that.

Reply:
Are you sure about that?
AFAICS, test_keyboard_interrupt shouldn't pass if signal mask is inherited.
Yet, it may be not true for Windows, IDK.

Reply:
Whoa, something funny happens with python 3.6 tests :)

Reply:
Ah, hm... I think you are right, since Popen does `exec`, which should reset signal handlers.

Reply:
Yes, you are absolutely right. Sorry, my os programming skills are getting rusty :sweat_smile: